src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -6181,6 +6204,68 @@ IOperation popTarget(IOperation originalTarget)
                         throw ExceptionUtilities.UnexpectedValue(originalTarget.Kind);
                 }
             }
+
+            static bool onlyContainsEmptyLeafNestedInitializers(IMemberInitializerOperation memberInitializer)
+            {
+                return memberInitializer.Initializer is IObjectOrCollectionInitializerOperation initializer
+                    && initializer.Initializers.All(e => e is IMemberInitializerOperation assignment && onlyContainsEmptyLeafNestedInitializers(assignment));
+            }
+
+            void addIndexes(IMemberInitializerOperation memberInitializer)
+            {
+                var lhs = memberInitializer.InitializedMember;
+                // If we have an element access of the form `[arguments] = { ... }`, we'll evaluate `arguments` only
+                if (lhs is IPropertyReferenceOperation propertyReference)
+                {
+                    foreach (var argument in propertyReference.Arguments)
+                    {
+                        if (argument is { ArgumentKind: ArgumentKind.ParamArray })
+                        {
+                            var array = (IArrayCreationOperation)argument.Value;
```

Comment:
>IArrayCreationOperation [](http://example.com/codeflow?start=41&length=23)

Consider incorporating this as a type check into the ```if``` condition in order to make the code more robust to future language changes.

Reply:
The thread was marked as addressed, but it doesn't look like the suggestion was implemented. First, from the perspective of  "making the code more robust to future language change", the code remains fragile. It is still going to crash if the value is not ```IArrayCreationOperation```. Whereas, we could simply go to the else branch instead, which would be the right thing to do for params collection case, for example. Second, our general guidance is to not use `as` cast if we don't expect the cast to fail, use a regular cast. If `as` cast is used, it should be followed by a null check.